system/screen.h

```diff
@@ -46,9 +46,21 @@ typedef enum {
 } screen_colour_t;
 
 /**
- * BIOS/UEFI(GOP) agnostic framebuffer copy
+ * Colour Palette definition
  */
+typedef struct {
+    screen_colour_t background;
+    screen_colour_t foreground;
+    screen_colour_t title_background;
+    screen_colour_t title_foreground;
+    screen_colour_t footer_background;
+    screen_colour_t footer_foreground;
+    screen_colour_t popup_background;
+} screen_palette_t;
```

Comment:
I'm not sure if "screen_palette" - in participial palette, may be the best name, perhaps more like "color_template"? After all, we are not changing the color palette -  it stays the same, with the "classic" 15+1 VGA colors.